apps/web/src/app/components/CollateralTypeSwitcher/CollateralTypeList.tsx

```diff
@@ -1,207 +1,164 @@
-
-  &:first-of-type {
-    text-align: left;
-  }
-`;
-
-const CollateralTypesGridItem = styled.div`
-  text-align: right;
-  font-size: 14px;
-  padding: 20px 0 20px 10px;
-  cursor: pointer;
-
-  &:first-of-type {
-    text-align: left;
-    padding: 20px 0;
-  }
-`;
-
-const AssetInfo = styled.div`
-  display: flex;
-  position: relative;
-  align-items: center;
-  padding-left: 40px;
-
-  img {
-    position: absolute;
-    left: 0;
-  }
-
-  @media screen and (max-width: 350px) {
-    padding-left: 0;
-    img {
-      display: none;
-    }
-  }
-`;
-
-const GridWrap = styled.div`
-  max-height: 305px;
-  overflow-y: auto;
-  padding: 0 25px;
-`;
-
-const CollateralTypeList = ({ width, setAnchor, anchor, ...rest }) => {
+import { useLoanActionHandlers, useSetLoanRecipientNetwork } from 'store/loan/hooks';
+import { Trans } from '@lingui/macro';
+import { BalanceAndValueWrap, DashGrid, HeaderText, List, walletBreakpoint } from '../Wallet/styledComponents';
+import SingleChainItem from './SingleChainItem';
+import MultiChainItem from './MultiChainItem';
+import { CollateralTab } from './CollateralTypeListWrap';
+import { ICON_XCALL_NETWORK_ID } from 'constants/config';
+import SingleChainItemOverview from './SingleChainItemOverview';
+import { xChainMap } from 'app/pages/trade/bridge/_config/xChains';
+import { Typography } from 'app/theme';
+import { XChainId } from 'app/pages/trade/bridge/types';
+
+const CollateralTypeList = ({
+  setAnchor,
+  collateralTab,
+  query,
+}: {
+  setAnchor: (anchor: HTMLElement | null) => void;
+  collateralTab: CollateralTab;
+  query: string;
+}) => {
   const changeCollateralType = useCollateralChangeCollateralType();
-  const allCollateralData = useAllCollateralData();
-  const hideCollateralInfoColumn = useMedia('(max-width: 500px)');
-  const handleEscape = useKeyPress('Escape');
+  const setLoanNetwork = useSetLoanRecipientNetwork();
+  const changeCollateralXChain = useChangeCollateralXChain();
   const { onAdjust: adjust } = useCollateralActionHandlers();
   const { onAdjust: adjustLoan } = useLoanActionHandlers();
+  const isSmallScreen = useMedia(`(max-width: ${walletBreakpoint})`);
+
+  const { data: allPositionsData } = useXCollateralDataByToken();
+  const { data: allCollateralData } = useAllCollateralData();
 
   const handleCollateralTypeChange = useCallback(
-    symbol => {
+    (symbol: string, xChainId: XChainId) => {
       setAnchor(null);
       changeCollateralType(symbol);
       adjust(false);
       adjustLoan(false);
+      if (xChainId) {
+        changeCollateralXChain(xChainId);
+        setLoanNetwork(xChainId);
+      }
     },
-    [changeCollateralType, setAnchor, adjust, adjustLoan],
+    [changeCollateralType, setAnchor, adjust, adjustLoan, changeCollateralXChain, setLoanNetwork],
   );
 
-  useEffect(() => {
-    if (anchor && handleEscape) {
-      setAnchor(null);
-    }
-  }, [anchor, handleEscape, setAnchor]);
+  const filteredPositions = useMemo(() => {
```

Comment:
after you unify two types, refactor the code. 

positions = collateralTab === CollateralTab.YOUR ? allPositionsData(userPositionData) + allCollateralData(allPositionData)
const filteredPositions = ....

Reply:
moved to https://github.com/balancednetwork/balanced-network-interface/issues/1388